Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-filter/ContestFilter.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback, useMemo, useState } from 'react';
+
+import Heading, { HeadingType } from '../../guidelines/headings/Heading';
+import List, { Orientation } from '../../guidelines/lists/List';
+import { FilterType, IFilter } from '../../../common/contest-types';
+import ExpandButton from '../../guidelines/buttons/ExpandButton';
+import Button, { ButtonSize, ButtonType } from '../../guidelines/buttons/Button';
+import concatClassNames from '../../../utils/class-names';
+
+import styles from './ContestFilter.module.scss';
+
+
+interface IContestFilterProps {
+    values: IFilter[];
+    type: FilterType;
+    onSelect: (filterId: number) => void;
+    maxDisplayCount: number;
+}
+
+const ContestFilter = ({
+    values,
+    type,
+    onSelect,
+    maxDisplayCount,
+}: IContestFilterProps) => {
+    const initialExpanded = false;
+    const [ expanded, setExpanded ] = useState(initialExpanded);
+
+    const listOrientation = useMemo(
+        () => type === FilterType.Status
+            ? Orientation.horizontal
+            : Orientation.vertical,
+        [ type ],
+    );
+    
+    const filtersToDisplayCount = useMemo(
+        () => expanded
+            ? values.length
+            : maxDisplayCount,
+        [ expanded, values, maxDisplayCount ],
+    );
+    
+    const toggleFiltersExpanded = useCallback(
+        (isExpanded: boolean) => {
+            setExpanded(isExpanded);
+        },
+        [],
+    );
+
+    const renderExpandButton = useCallback(
+        (allFilters: IFilter[]) => allFilters.length > maxDisplayCount
+            ? <ExpandButton onExpandChanged={toggleFiltersExpanded}/>
+            : null,
+        [ toggleFiltersExpanded, maxDisplayCount ],
+    );
+
+    const getRenderFilterItemFunc = useCallback(
+        (filterType: FilterType) => ({ id, name }: IFilter) => {
+            // TODO: investigate why filters change ids
+            //  and use id instead of name for checking if filter is selected
```

Comment:
This comment is outdated. Please, delete it

Reply:
Done.